dedupe/ckan_api.py

```diff
@@ -83,7 +83,7 @@ def request(self, method, path, **kwargs):
     def get(self, path, **kwargs):
         return self.request('GET', path, **kwargs)
 
-    def get_oldest_dataset(self, organization_name, identifier, is_collection):
+    def get_dataset_to_save(self, organization_name, identifier, is_collection, oldest=True):
```

Comment:
Minor, but this feels like you're pushing the logic of "which dataset do I save ?" into the ckan_api module. ckan_api should only be concerned with providing an abstraction for the API. You could imagine a more complicated heuristic to determine which dataset is the correct one where we inspect different fields or harvest objects. That heuristic would make more sense to live in the Deduper class.

Maybe this should just be `get_dataset(self, organization_name, identifier, is_collection, sort_order='asc')`.